File: Competitive-Prog/ ACM U of T Tryouts C1 A - Rock Paper Scissors Fox /COCI '06 Contest 6 #1 Prase/sub.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
#include <string>
using namespace std;

int main() {
    unordered_map<string, string> rpsfGame;
    rpsfGame["Rock"] = "Paper";
    rpsfGame["Paper"] = "Scissors";
    rpsfGame["Scissors"] = "Rock";
    rpsfGame["Fox"] = "Foxen";

    int N;
    cin >> N;
    cin.ignore(); // ignore newline character

    vector<string> inps(N);
    for (int s = 0; s < N; s++) {
        getline(cin, inps[s]);
    }

    vector<string> outs;
    for (string i : inps) {
        if (i.compare("Foxen") == 0) {
            break;
        } else {
            outs.push_back(rpsfGame[i]);
        }
    }

    for (string j : outs) {
        cout << j << endl;
    }

    return 0;
}

File: Competitive-Prog/ ACM U of T Tryouts C1 A - Rock Paper Scissors Fox / ACM U of T Tryouts C1 A - Rock Paper Scissors Fox /Sub.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) {
       Scanner key= new Scanner(System.in);
        HashMap<String,String> rpsfGame= new HashMap<>();

        rpsfGame.put("Rock","Paper");
        rpsfGame.put("Paper","Scissors");
        rpsfGame.put("Scissors","Rock");
        rpsfGame.put("Fox","Foxen");

        int N= key.nextInt();
        key.nextLine();

        String [] inps= new String[N];
        for(int s= 0; s< inps.length; s++){
            inps[s]= key.nextLine();
        }
        ArrayList <String> outs= new ArrayList<>();
        for (String i : inps){

            if (i.compareTo("Foxen")==0){
                break;
            }
            else{
                outs.add(rpsfGame.get(i));
            }
        }

        for (String j: outs){
            System.out.println(j);
        }
        
        
    }
}
